samples/Basic_Delegates/app/application.cpp

```diff
@@ -1,45 +1,125 @@
-		ledInterval = reqInterval;
+		if (reqInterval <= 0) {
+			return false;
+		}
+		taskInterval = reqInterval;
 		return true;
 	}
-	void blink(bool reqRun) {
-		if (reqRun) {
-			ledTimer.initializeMs(ledInterval, TimerDelegate(&LedBlinker::ledBlink,this)).start();
-		}
-		else {
-			ledTimer.stop();
-		}
+
+	// This example show the way delegates have been used in Sming in the past.
+	void blinkOldDelegate() {
+		taskTimer.initializeMs(taskInterval, TimerDelegate(&Task::doOldDelegate, this)).start();
 	}
-	void ledBlink () { ledState = !ledState ; digitalWrite(ledPin, ledState);}
 
-private :
-	int ledPin = 2;
-	Timer ledTimer;
-	int ledInterval = 1000;
-	bool ledState = true;
-};
+	// This example shows how to use a plain old ordinary function as a callback
+	void callPlainOldOrdinaryFunction() {
+		taskTimer.initializeMs(taskInterval, TimerDelegateStdFunction(plainOldOrdinaryFunction)).start();
+		// or just
+		// taskTimer.initializeMs(taskInterval, plainOldOrdinaryFunction).start();
+	}
+
+
+	// This example shows how to use std::bind to make us of a function that has more parameters than our signature has
+	void showHowToUseBind() {
+		auto b = std::bind(functionWithMoreComlicatedSignature, 2, "parameters");
+		taskTimer.initializeMs(taskInterval, b).start();
+	}
+
+	// Sming now allows the use of std::function
+	// This example shows how to use a lamda expression as a callback
+	void callLamda() {
+		int foo = 123;
+		taskTimer.initializeMs(taskInterval,
+			[foo]		// capture just foo by value (Note it would be bad to pass by reference as foo would be out of scope when the lamda function runs later)
+		()			// No parameters to the callback
+			-> void		// Returns nothing
+		{
+			if (foo == 123) {
+				debugf("lamda Callback foo is 123");
+			}
+			else
+			{
+				debugf("lamda Callback foo is not 123, crikey!");
+			}
+		})
+			.start();
+	}
 
-#define LEDPIN_1  2 // GPIO2
-#define LEDPIN_2  4 // GPIO4
 
-LedBlinker myLed1 = LedBlinker(LEDPIN_1);
-LedBlinker myLed2 = LedBlinker(LEDPIN_2);
+
+	// This example shows how to use a member function as a callback
+	void callMemberFunction() {
+
+		// A non-static member function must be called with an object. 
+		// That is, it always implicitly passes "this" pointer as its argument.
+		// But because our callback specifies that we don't take any arguments (<void(void)>), 
+		// you must use std::bind to bind the first (and the only) argument.
+
+		TimerDelegateStdFunction b = std::bind(&Task::callbackMemberFunction, this);
+		taskTimer.initializeMs(taskInterval, b).start();
+	}
+
+	void doOldDelegate()
+	{
+		debugf("doOldDelegate");
+	}
+	void callbackMemberFunction()
+	{
+		debugf("callMemberFunction");
+	}
+
+
+private:
+	Timer taskTimer;
+	int taskInterval = 1000;
+
+};
+
+Task task1;
+Task task2;
+Task task3;
+Task task4;
+Task task5;
 
 void init()
 {
-	myLed1.setTimer(1000);
-	myLed1.blink(true);
-	myLed2.setTimer(500);
-	myLed2.blink(true);
+	WifiStation.enable(false);
```

Comment:
Remove the Wifi* call on this line and the one below. They are not directly related to the example and to what you are trying to demonstrate.